components/collator/src/comparison.rs

```diff
@@ -69,11 +69,35 @@ pub struct Collator {
 
 impl Collator {
     /// Instantiates a collator for a given locale with the given options
-    ///
-    /// [📚 Help choosing a constructor](icu_provider::constructors)
-    /// <div class="stab unstable">
-    /// ⚠️ The bounds on this function may change over time, including in SemVer minor releases.
-    /// </div>
+    #[cfg(feature = "data")]
+    pub fn try_new(locale: &DataLocale, options: CollatorOptions) -> Result<Self, CollatorError> {
+        Self::try_new_unstable_internal(
+            &crate::provider::Baked,
+            DataPayload::from_static_ref(
+                icu_normalizer::provider::Baked::SINGLETON_NORMALIZER_NFD_V1,
+            ),
+            DataPayload::from_static_ref(
+                icu_normalizer::provider::Baked::SINGLETON_NORMALIZER_NFDEX_V1,
+            ),
```

Comment:
Nit: For nested constructors I think we said that we wanted to call the globaldata constructor from the other crate rather than linking the data from that crate into this crate.

Reply:
I don't know if there's a public API we can go, at the moment this crate directly uses the `DataPayload`s.

I think what we decided was that if a crate calls a constructor from another crate, the data constructor should call other data constructors directly. This is not quite what's happening here.

Reply:
Still a bit uneasy; should we refactor the code to hold a Normalizer instead of the payloads, or perhaps create a Normalizer than then use a doc(hidden) API to extract the payloads from it? Let's discuss

Reply:
Discussion:

- If they are in different codegen units, it's possible that the data may be anchored in two different places (duplicated)
- But LTO or codegen-units=1 should fix it in most cases
- Most of the call sites are const constructors which have the same problem, but it will show up in client code rather than ICU4X code

Conclusion:

- Land this as-is and fix it later if we find it is an actual problem